.github/workflows/run-k8s-tests-on-aks.yaml

```diff
@@ -27,83 +27,45 @@ jobs:
           - host_os: cbl-mariner
             vmm: clh
     runs-on: ubuntu-latest
+    env:
+      DOCKER_REGISTRY: ${{ inputs.registry }}
+      DOCKER_REPO: ${{ inputs.repo }}
+      DOCKER_TAG: ${{ inputs.tag }}
+      GH_PR_NUMBER: ${{ github.event.pull_request.number }}
+      GH_PR_SHA: ${{ github.event.pull_request.head.sha }}
+      KATA_HOST_OS: ${{ matrix.host_os }}
+      KATA_HYPERVISOR: ${{ matrix.vmm }}
     steps:
       - uses: actions/checkout@v3
         with:
           ref: ${{ github.event.pull_request.head.sha }}
 
       - name: Download Azure CLI
-        run: |
-          curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
-          # The aks-preview extension is required while the Mariner Kata host is in preview.
-          az extension add --name aks-preview
+        run: bash tests/integration/gha-run.sh install-az-cli
+        env:
+          AZ_APPID: ${{ secrets.AZ_APPID }}
+          AZ_PASSWORD: ${{ secrets.AZ_PASSWORD }}
+          AZ_TENANT_ID: ${{ secrets.AZ_TENANT_ID }}
```

Comment:
These are scoped to the wrong step (the next one does the login), GHA has this weird "feature" that env assignments leak to following steps 😃

Reply:
Oops, fixed!

> GHA has this weird "feature" that env assignments leak to following steps 😃

Hmm have you seen this in the wild? Can't find any docs on this.

Reply:
there are no docs on this, but i'm fairly sure i've seen this happen.